crates/core/src/types/receipt.rs

```diff
@@ -54,11 +58,47 @@ impl RLPEncode for Receipt {
     }
 }
 
+impl RLPDecode for Receipt {
+    fn decode_unfinished(rlp: &[u8]) -> Result<(Self, &[u8]), RLPDecodeError> {
```

Comment:
why unfinished?

Reply:
The `RLPDecode` trait provides the implementation for `decode`, but you have to implement `decode_unfinished` on your type